src/main/java/com/imsweb/x12/Separators.java

```diff
@@ -88,6 +97,24 @@ public String toString() {
         return "[" + _segment + "," + _element + "," + _composite + "]";
     }
 
+    private void setSeparatorPatterns(Character segment, Character element, Character composite) {
+        _segmentPattern = Pattern.compile(Pattern.quote(segment.toString()));
+        _elementPattern = Pattern.compile(Pattern.quote(element.toString()));
+        _compositePattern = Pattern.compile(Pattern.quote(composite.toString()));
+    }
+
+    public String[] splitElement(String line) {
+        return (line != null && _elementPattern != null) ? _elementPattern.split(line) : null;
```

Comment:
Quick question.  Why are you testing is `_elementPattern` is not null when I can't see any way it could be.  It doesn't hurt to test, but both constructors set it up so I don't see how it would be null.  Same for other 2 patterns.

Reply:
There is a constructor that takes the 3 separators as parameters. That will let you create a Separator with all 3 separators being null. I don't see why this would ever happen in practice but if someone did do that then I set the separator patterns to null.

Reply:
I guess you are right, but that should probably not be allowed.  We can leave this as is now.